doc/cookbook/example_developer_workflow.md

```diff
@@ -1,49 +1,74 @@
+development practices, including
+the `--build` and `--push-images` flags to the
+`pachctl update pipeline` command that enable you to
+build and push images to a Docker registry.
 
+The following diagram demonstrates basic Pachyderm
+development workflow:
 
-## Basic workflow
+![Developer Workflow](../images/d_developer_workflow102.svg)
 
-![alt tag](developer_workflow.png)
+The automated developer workflow includes the following steps:
 
-As you write code, you test it in containers and notebooks against sample data in Pachyderm repos.  You can also run your code in development pipelines in Pachyderm.  Pachyderm provides facilities to help with day-to-day development practices, including the ``--build`` and ``--push-images`` flags to the ``update pipeline`` command, which can build & push or just push images to a docker registry.
+1. A new commit triggers a Git hook.
 
-There are a couple of things to note about the files shown in git, in the left-hand side of the diagram above.  The pipeline.json template file, in addition to being used for CI/CD as noted below,  could be used with local build targets in a makefile for development purposes: the local build uses DOCKERFILE and creates a pipeline.json for use in development pipelines.  This is optional, of course, but may fit in with some workflows.
+   Typically, Pachyderm users store the following artifacts in a
+   Git repository:
 
-Once your code is ready to commit to your git repo, here are the steps that can form the basis of a production workflow.
+   - A Dockerfile that you use to build local images.
+   - A `pipeline.json` specification file that you can use in a `Makefile` to
+   create local builds, as well as in the CI/CD workflows.
+   - The code that performs data transformations.
 
-### 1. Git commit hook
+   A [commit hook in Git](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
+   for your repository triggers the CI/CD process. It uses the
+   information in your pipeline specification for subsequent steps.
 
-A [commit hook in git](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) for your kicks off the
-Continous Integration/Continuous Deployment process.  It should  use the information present in a template for your Pachyderm pipelines for subsequent steps.
+2. Build an image
 
-### 2. Build image
+   Your CI process must automatically start the build of a Docker container
+   image based on your code and the Dockerfile.
 
-Your CI process should automatically kick off the build of an docker container image based on your code and the DOCKERFILE. That image will be used in the next step.
+3. Push the image tagged with commit ID to an image registry.
 
-### 3. Push to registry tagged with commit id
+   Your CI process pushes a Docker image created in Step 2 to your preferred
+   image registry. When you build your image by using  the `--built` flag with
+   the `pachctl update pipeline` command, Pachyderm tags the image with the
```

Comment:
> the `pachctl update pipeline` command, Pachyderm tags the image with the
   Git commit SHA.

This is incorrect. Best practices in CI is to use the commit sha. --build just gives it a random UUID tag to the image. 

**Also, your mixing up two concepts here -- you can't use --build AND use CI. Those are doing the same thing.** The `--build` flag is a flag to `pachctl update pipeline` so I'm not sure why that's being mentioned here. We can discuss in TOH or chat with @pappasilenus , but these are two distinct options. 

1. Users can develop locally and then use `update pipeline --build` which will build the image, tag it with a random UUID, push to a registry, AND update the pipeline to use that new image. This works great for the individual developer workflow because its one cmd that does it all. That said, it also requires your data scientists to have docker knowledge and have it all installed and running. It also is more limiting because it's all being done by pachyderm so there is less functionality/control than using CI.

OR

2. Use Ci to automate pipeline updates. Data scientist just pushes code to Git -- doesn't need to think about anything else. CI then does the above process here -- uses dockerfile to build, tag, and push container, and then update the json and makes an API call to pachyderm to update the pipeline. Generally speaking, using CI is much more heavy-weight process but it has a few advantages:
- data eng abstracts away lots fo details for data scientists who just want to push code and not think about all the docker/pachyderm details
- they can run other CI test on the code before pushing
- they have more control over naming/tagging schemes such as using the Git commit SHA
- probably others I'm not thinking of right now